test/unittests/test_parse_de.py

```diff
@@ -33,55 +37,218 @@ def tearDownModule():
 
 
 class TestNormalize(unittest.TestCase):
+
     def test_articles(self):
         self.assertEqual(
             normalize("dies ist der test", lang="de-de", remove_articles=True),
             "dies ist test")
         self.assertEqual(
             normalize("und noch ein Test", lang="de-de", remove_articles=True),
-            "und noch 1 Test")
+            "und noch ein Test")
         self.assertEqual(normalize("dies ist der Extra-Test", lang="de-de",
                                    remove_articles=False),
                          "dies ist der Extra-Test")
 
-    def test_extract_number(self):
-        self.assertEqual(extract_number("dies ist der 1. Test",
-                                        lang="de-de"), 1)
-        self.assertEqual(extract_number("dies ist der erste Test",
-                                        lang="de-de"), 1)
-        self.assertEqual(extract_number("dies ist 2 Test", lang="de-de"), 2)
-        self.assertEqual(extract_number("dies ist zweiter Test", lang="de-de"),
-                         2)
+    def test_spaces(self):
+        self.assertEqual(normalize("  dies   ist  ein    test", lang="de-de"),
+                         "dies ist ein test")
+        self.assertEqual(normalize("  dies   ist  ein    test  ",
+                                   lang="de-de"), "dies ist ein test")
+
+    def test_numbers(self):
+        self.assertEqual(
+            normalize("dies ist eins zwei drei test", lang="de-de"),
+            "dies ist 1 2 3 test")
+        self.assertEqual(
+            normalize("es ist vier fünf sechs test", lang="de-de"),
+            "es ist 4 5 6 test")
+        self.assertEqual(
+            normalize("es ist sieben acht neun test", lang="de-de"),
+            "es ist 7 8 9 test")
+        self.assertEqual(
+            normalize("es ist sieben acht neun test", lang="de-de"),
+            "es ist 7 8 9 test")
+        self.assertEqual(
+            normalize("dies ist zehn elf zwölf test", lang="de-de"),
+            "dies ist 10 11 12 test")
         self.assertEqual(
-            extract_number("dies ist der dritte Test", lang="de-de"), 3)
+            normalize("dies ist dreizehn vierzehn test", lang="de-de"),
```

Comment:
"dies ist dreizehn vierzehn test" isn't a really german sentence. Is this meant to be grammatically wrong? I'd say: "dies ist der dreizehnte vierzehnte test".

Reply:
i could have written "dreizehn vierzehn mal falsch geschrieben" :P. The normalizer doesn't deal with ordinals